pkg/addon/agent/controller/token.go

```diff
@@ -82,86 +83,119 @@ func (r *TokenReconciler) Reconcile(ctx context.Context, request reconcile.Reque
 		return reconcile.Result{}, errors.Wrapf(err, "failed to ensure service account")
 	}
 
+	expiring, err := r.sync(ctx, managed)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrapf(err, "failed to sync token")
+	}
+
+	var conditions []metav1.Condition
+	now := metav1.Now()
+	tokenRefreshTime := now
+
+	if expiring == nil {
+		logger.Info("Skipped creating token")
+		// token is not refreshed, keep the previous expiration timestamp
+		expiring = managed.Status.ExpirationTimestamp
+		if managed.Status.TokenSecretRef != nil {
+			tokenRefreshTime = managed.Status.TokenSecretRef.LastRefreshTimestamp
+		}
+	} else {
+		// token is refreshed, update the token reported condition last transition time
+		conditions = append(conditions, metav1.Condition{
+			Type:               authv1beta1.ConditionTypeTokenReported,
+			Status:             metav1.ConditionTrue,
+			Reason:             "TokenReported",
+			LastTransitionTime: now,
+		})
+	}
+
+	// after sync func succeeds, the secret must exist, add the conditions if not exist
+	if meta.FindStatusCondition(managed.Status.Conditions, authv1beta1.ConditionTypeSecretCreated) == nil {
```

Comment:
I think you can use the helper in meta to directly set the condition?

Reply:
I think these 2 conditions are similar, can we only keep one and delete another one?

Reply:
@qiujian16 ptal